src/main/java/com/wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,61 @@
 package com.wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import com.wolox.training.exceptions.IdMismatchException;
+import com.wolox.training.exceptions.NotFoundException;
+import com.wolox.training.models.Book;
+import com.wolox.training.repositories.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.*;
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public Iterable<Book> getAllBooks(){
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id){
+        return bookRepository.findById(id).orElseThrow(NotFoundException::new);
+    }
+
+    @GetMapping("/author/{authorName}")
+    public Book findByAuthor(@PathVariable String authorName){
+        return bookRepository.findByAuthor(authorName).orElseThrow(NotFoundException::new);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
```

Comment:
This is just extra information: you could add a response status with the NO_CONTENT code (204) to inform that the operation was done correctly and no content should be returned.